File: src/main/scala-2.11/com/lei/leetcode/P605/Solution.java
package com.lei.leetcode.P605;

/**
 * Created by Lei on 2017/7/13.
 */
public class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int index = 0;
        int len = flowerbed.length;

        while (index < len && n != 0) {
            if (flowerbed[index] == 0 && index == 0 && flowerbed[index + 1] == 0) {
                flowerbed[index] = 1;
                n -= 1;
            } else if (flowerbed[index] == 0 && flowerbed[index - 1] == 0 && index == len - 1) {
                flowerbed[index] = 1;
                n -= 1;gst
            } else if ([flowerbed[index] == 0 && flowerbed[index - 1] == 0 && flowerbed[index + 1] == 0){
                flowerbed[index] = 1;
                n -= 1;
            }
            index += 1;
        }

        if (n == 0) return true;

        else return false;
    }
}

File: src/main/scala-2.11/com/lei/leetcode/P605/Solution.scala
package com.lei.leetcode.P605

/**
  * Created by Lei on 2017/7/13.
  */
object Solution extends App {

  def canPlaceFlowers(flowerbed: Array[Int], n: Int): Boolean = {
    var index = 0
    var left = n
    val length = flowerbed.length
    while (index < length && left != 0) {
      if (flowerbed(index) == 0 && index == 0 && flowerbed(index + 1) == 0) {
        flowerbed(index) = 1
        left -= 1
      } else if (flowerbed(index) == 0 && flowerbed(index - 1) == 0 && index == length - 1) {
        flowerbed(index) = 1
        left -= 1
      } else if (flowerbed(index) == 0 && flowerbed(index - 1) == 0 && flowerbed(index + 1) == 0) {
        flowerbed(index) = 1
        left -= 1
      }
      index += 1
    }

    if (left == 0) true else false
  }

  val f = Array(1, 0, 0, 0, 1)

  println(canPlaceFlowers(f, 1))

}
